substrate-archive/src/tasks.rs

```diff
@@ -221,12 +221,12 @@ where
 			hash,
 			number,
 		};
-		// We destroy the Arc and transform the Mutex here in order to avoid additional allocation.
-		// The Arc is cloned into the thread-local tracing subscriber in the scope of `storage`, creating
-		// 2 strong references. When block execution finishes, storage is collected and that reference is dropped.
-		// This allows us to unwrap it here. QED.
-		let traces = Arc::try_unwrap(span_events).map_err(|_| TracingError::NoTraceAccess)?.into_inner();
-		let traces = Traces::new(number.into(), hash.as_ref().to_vec(), traces.events, traces.spans);
+
+		let mut traces = span_events.lock();
+		let spans = traces.spans.drain(..).collect::<Vec<SpanMessage>>();
+		let events = traces.events.drain(..).collect::<Vec<EventMessage>>();
```

Comment:
This looks similar to how [we did it in substrate](https://github.com/paritytech/substrate/blob/20f1e369edb860a8287d79dd85a27b50e391b07c/client/tracing/src/block/mod.rs#L253-L268).

Reply:
Yeah, in general this implementation brings trace_block stuff in archive closer to what the code looks like in substrate :P